File: leetcode2/0374. Guess Number Higher or Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left=1
        right=n
        while right>=left:
            mid=(left+right)//2
            if guess(mid)==0:
                return mid
            elif guess(mid)==-1:
                right=mid-1
            else:
                left=mid+1

File: C#/0374. Guess Number Higher or Lower.cs
// Forward declaration of guess API.
// @param num, your guess
// @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        int left=1;
        int right=n;
        while(left<=right){
            int mid=left+(right-left)/2;
            int var=guess(mid);
            if (var==0){return mid;}
            else if(var==-1){right=mid-1;}
            else{left=mid+1;}
        }
        return 0;
    }
};
